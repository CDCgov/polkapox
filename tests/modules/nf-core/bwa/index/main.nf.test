nextflow_process {

    name "Test Process BWA_INDEX"
    script "modules/nf-core/bwa/index/main.nf"
    process "BWA_INDEX"

    test("Test | Clade I") {

        when {
            params {
                clade = 'cladeI'
            }
            process {
                """
                // define inputs of the process here. Example:
                 input[0] = file("/assets/NC_003310_clade_I.fasta")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
        test("Test | Clade II") {

        when {
            params {
                clade = 'cladeII'
            }
            process {
                """
                // define inputs of the process here. Example:
                 input[0] = file("/assets/MPXV-UK_P2.noN.fasta")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
