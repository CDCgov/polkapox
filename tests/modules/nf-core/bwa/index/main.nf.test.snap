{
    "Test | Clade II/FASTAonly": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "index": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-24T16:40:06.832684709"
    },
    "Test | Clade I": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "index": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-11T16:15:32.769447309"
    },
    "Test | Clade I/FASTAonly": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "index": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-24T16:39:02.534059762"
    },
    "Test | Clade II": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "index": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-11T16:15:39.849326942"
    }
}